### DISCUSSION TITLE
The rules regarding --sentence and ellipses

### DISCUSSION THREAD

**Discussion by despairTK** [Purfview/whisper-standalone-win#193] [Ideas]
When a sentence ends with a "..." punctuation mark, it is automatically merged with the next sentence.
For example this situation:

177
00:24:45,760 --> 00:24:48,300
So I think...

178
00:24:48,300 --> 00:24:49,940
hmm...

179
00:24:49,940 --> 00:24:53,560
we can add like a cover page for our cafe design.

After merging:

177
00:24:45,760 --> 00:24:53,560
So I think...hmm...we can add like a cover page for our cafe design.

**Comment by Purfview** [comment]
Share json and srt of it.

**Reply by despairTK** [reply]
[test.zip](https://github.com/Purfview/whisper-standalone-win/files/14317772/test.zip)

I transcribed it again and found that my example sentences were automatically merged again, but sentences elsewhere in the subtitles were still unmerged. You can try searching with the "..." keyword.

**Reply by Purfview** [reply]
Ar you sure it was without spaces like that -> "think...hmm...we"?

So it splits on "..." in your shared files.
I think it shouldn't split because in the first post and shared test the ellipses doesn't mean the end of the sentences.

**Reply by despairTK** [reply]
I don’t know why the example I mentioned above disappeared in the second transcription, but I found another one, as follows:

79
00:15:21,500 --> 00:15:28,860
So if it's okay if you don't follow how I'm layouting the posters here, because they're in...

80
00:15:28,860 --> 00:15:30,940
just in one size, so yeah.

**Reply by Purfview** [reply]
That is exactly what you've shared before.

**Reply by despairTK** [reply]
It's possible that I expressed myself incorrectly. What I meant was. There are some transcription results that end in "..." Punctuation marks ending in breaks that should normally be merged with the sentence that follows.

**Reply by Purfview** [reply]
So, you want the ellipse merged in test.zip?

**Reply by despairTK** [reply]
I think when the --sentence parameter is used. End with "..." Individual sentences ending in "..." should be merged with the next sentence, like the example given at the very beginning of this thread.

"..." The presence of this symbol generally means that a sentence is not finished, just that the speech interval is lengthened.

**Reply by Purfview** [reply]
Now it's clear what you want.

> "..." The presence of this symbol generally means that a sentence is not finished

Not always, sometimes it's the end of sentence.

**Reply by Purfview** [reply]
Test if this works how you want with the ellipses -> https://we.tl/t-gYPJoBpFgw

UPDATE:
test2 [won't merge if ellipse is at the end of a segment] -> https://we.tl/t-5xaHxt0oYl

**Reply by despairTK** [reply]
Thanks for your work, I tried both test versions, the first test version is very effective, the second test version is not merged correctly.

**Reply by Purfview** [reply]
> the second test version is not merged correctly

share json and srt

**Reply by despairTK** [reply]
Files for the first test version:
[test01-01.zip](https://github.com/Purfview/whisper-standalone-win/files/14326254/test01-01.zip)
[test01-02.zip](https://github.com/Purfview/whisper-standalone-win/files/14326255/test01-02.zip)

Files for the second test version:
[test02-01.zip](https://github.com/Purfview/whisper-standalone-win/files/14326256/test02-01.zip)
[test02-02.zip](https://github.com/Purfview/whisper-standalone-win/files/14326257/test02-02.zip)


I just did a few more tests and found that the second test version will work normally, but sometimes it will encounter some special circumstances.

For example, in test01-01:

1
00:00:00,000 --> 00:00:07,320
So now we reach the last section of chapter 1, and here we finally get to do what you all signed up for, which is... Draw.

Compare test01-02

1
00:00:00,000 --> 00:00:06,420
So now we reach the last section of chapter 1, and here we finally get to do what you all signed up for, which is...

2
00:00:07,020 --> 00:00:07,320
Draw.

If "Draw" is used as a single line of subtitles, it may be translated into a meaning inconsistent with the content. For example, this subtitle file introduces the content of painting, so "Draw" should be translated into painting. But because it is disconnected from the previous sentence, the meaning after machine translation becomes drawing rewards.

However, this situation is relatively rare, and for me the first test version is more suitable for me.

**Reply by Purfview** [reply]
> I tried both test versions, the first test version is very effective, the second test version is not merged correctly.

Looking at your files, test2 works correctly, when test1 is not.
So, I'll go with the test2 version.
If you think that test1 is better for you then I can add option to turn on the test1 behavior.

>but sometimes it will encounter some special circumstances.

Model put "Draw" in the new segment with the capital letter, so it's a mistake of a model not a mistake of "..." splitting.
If "Draw" would be still in the same segment [in pure whisper output] then test2 wouldn't split it.

**Reply by despairTK** [reply]
Please follow the test2 version. This special situation I encountered rarely happens. Generally, when checking the subtitles, I will basically discover and modify it.

**Comment by Lycoan** [comment]
> Now it's clear what you want.
> 
> > "..." The presence of this symbol generally means that a sentence is not finished
> 
> Not always, sometimes it's the end of sentence.

Yes, sometimes it can be the end of the sentence: it can also indicate that the speaker leaves the scene and his voice fades.
If it is the end of the sentence, merging can lead to a wrong result, because the next part can belong to new speaker. That is why I like breaking subtitle by sentences, that way there can be no such merging error.

**Reply by Purfview** [reply]
> That is why I like breaking subtitle by sentences, that way there can be no such merging error.

But in his both examples sentence is broken on every "...".

**Reply by Purfview** [reply]
> that way there can be no such merging error.

He expressed it wrongly, in the OP he wrote a behavior he wishes [to merge on ellipses].

**Reply by Lycoan** [reply]
Okay, thank you for clarifying, now I know what  you meant by ellipses. :)
It will be a nice feature.

**Comment by Purfview** [comment]
Implemented in r172.4.
